src/main/java/edu/tamu/app/service/AbstractCatalogService.java

```diff
@@ -97,15 +103,47 @@ public void setProtocol(String protocol) {
     }
 
     protected String getAPIBase() {
-        return getProtocol() + "://" + getHost() + ":" + getPort() + "/" + getApp() + "/";
+        StringBuilder builder = new StringBuilder();
+
+        if (StringUtils.isNotEmpty(getProtocol())) {
+          builder.append(getProtocol());
+          builder.append(":");
+        }
+
+        builder.append("//");
+        builder.append(getHost());
+
+        if (StringUtils.isNotEmpty(getPort())) {
+          builder.append(":");
+          builder.append(getPort());
+        }
+
+        if (StringUtils.isNotEmpty(getApp())) {
+          builder.append("/");
+          builder.append(getApp());
```

Comment:
We need the trailing slash here.

Reply:
Actually, I think the final slash needs to be after this last if block, not in it.

Reply:
We end up with malformed voyager urls without the trailing slash: "[server]/vxwsrecord/217815/?view=full"